test/extended/util/client.go

```diff
@@ -337,6 +345,40 @@ func (c *CLI) SetupProject() string {
 	return newNamespace
 }
 
+func (c *CLI) setupNamespace() string {
+	requiresTestStart()
+	newNamespace := names.SimpleNameGenerator.GenerateName(fmt.Sprintf("e2e-test-%s-", c.kubeFramework.BaseName))
+	c.SetNamespace(newNamespace).ChangeUser(fmt.Sprintf("%s-user", newNamespace))
+
+	nsObject := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: newNamespace,
+		},
+	}
+	framework.Logf("Creating namespace %q", newNamespace)
+	_, err := c.AdminKubeClient().CoreV1().Namespaces().Create(context.Background(), nsObject, metav1.CreateOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	c.kubeFramework.AddNamespacesToDelete(nsObject)
+
```

Comment:
you need to do the work of actually creating the things in teh standard openshfit project template like the proper annotations and rolebindings.

Reply:
A project creates role bindings for cluster roles `system:image-puller`, `system:image-builder` and `system:deployer`. The first two relate only to OCP APIs, so we can skip them.
The last one I will need it if I change the kubeconfig for the `UserConfig`, as it grants permissions to operate on pods, replication controllers, events and other OCP APIs.

Reply:
> The last one I will need it if I change the kubeconfig for the UserConfig, as it grants permissions to operate on pods, replication controllers, events and other OCP APIs.

It must switch users or the testing is invalid.  The annotations are important because they change the way that consumers are able to determine ownership.

Reply:
Done

Reply:
still missing annotations.  see https://github.com/openshift/openshift-apiserver/blob/master/pkg/project/apiserver/registry/projectrequest/delegated/sample_template.go#L46-L48